oauth2_http/javatests/com/google/auth/oauth2/MockMetadataServerTransport.java

```diff
@@ -62,6 +62,15 @@ public class MockMetadataServerTransport extends MockHttpTransport {
 
   private byte[] signature;
 
+  private String plaintextS2AAddressJsonKey;
+
+  private String plaintextS2AAddress;
+
+  private String mtlsS2AAddressJsonKey;
```

Comment:
Do the tests need to make the two JsonKeys configurable? Can they just be referenced from the constants from the S2A class?

Reply:
Ah nvm, I jumped the gun. I see the tests below where the data is put in the wrong key

Reply:
In `S2ATest`, we reference the constants in the `S2A` class. The reason I made the keys configurable was to test the logic added as part of this comment thread: https://github.com/googleapis/google-auth-library-java/pull/1400#discussion_r1815800047. 

For example, in `getS2AAddress_queryEndpointResponseInvalidPlaintextJsonKey_plaintextEmptyAddress`, I inject an invalid json key via `plaintextS2AAddressJsonKey`, to confirm that the `mtls_address` is still parsed, even though there's a problem parsing the `plaintext_address`.